v2/slogger/rolling_file_appender/rolling_file_appender.go

```diff
@@ -127,7 +160,10 @@ func (self *RollingFileAppender) Append(log *slogger.Log) error {
 		return err
 	}
 
-	if self.MaxFileSize > 0 && self.curFileSize > self.MaxFileSize {
+	if (self.maxFileSize > 0 && self.curFileSize > self.maxFileSize) ||
+		(self.maxDuration > 0 &&
+			self.state != nil &&
```

Comment:
That's cool. This line just made me think harder about what the `state` object is for and how it works, inspiring the comment below.